File: src/main/scala/com/wikibooks/spark/ch2/scala/RDDCreateSample.scala
package com.wikibooks.spark.ch2.scala

import org.apache.spark.{SparkConf, SparkContext}

object RDDCreateSample {

  def main(arr: Array[String]) {

    val conf = new SparkConf()
      .setMaster("local[*]")
      .setAppName("RDDCreateSample")

    val sc = new SparkContext(conf)

    // 2.1.3절 예제 2-6
    val rdd1 = sc.parallelize(List("a", "b", "c", "d", "e"))

    // 2.1.3절 예제 2-8
    val rdd2 = sc.textFile("<spark_home_dir>/README.md")

    println(rdd1.collect.mkString(", "))
    println(rdd2.collect.mkString(", "))

    sc.stop
  }
}

File: src/main/java/com/wikibooks/spark/ch2/RDDCreateSample.java
package com.wikibooks.spark.ch2;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

import java.util.Arrays;

public class RDDCreateSample {

  public static void main(String[] args) throws Exception {

    SparkConf conf = new SparkConf()
            .setMaster("local[*]")
            .setAppName("RDDCreateSample");

    JavaSparkContext sc = new JavaSparkContext(conf);

    // 2.1.3절 예제 2-6
    JavaRDD<String> rdd1 = sc.parallelize(Arrays.asList("a", "b", "c", "d", "e"));

    // 2.1.3절 예제 2-8
    JavaRDD<String> rdd2 = sc.textFile("<spark_home_dir>/README.md");

    for (String v : rdd1.collect()) {
      System.out.println(v);
    }

    for (String v : rdd2.collect()) {
      System.out.println(v);
    }

    sc.stop();
  }
}